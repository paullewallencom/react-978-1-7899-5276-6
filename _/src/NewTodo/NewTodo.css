.NewTodo {
  margin: 20px;
  padding: 20px;
  border: 2px solid #00ffd8;
  background: #ddffee;
  text-align: center;
}
