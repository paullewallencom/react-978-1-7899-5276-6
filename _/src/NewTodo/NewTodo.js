import React, { Component } from "react";
import { Button, Input, InputGroup } from "reactstrap";
import "./NewTodo.css";

class NewTodo extends Component {
  constructor(props) {
    super(props);
    this.state = { item: "" };
    this.handleUpdate = this.handleUpdate.bind(this);
    this.addTodo = this.addTodo.bind(this);
  }
  addTodo() {
    this.props.addTodo(this.state.item);
    this.setState({ item: "" });
  }
  handleUpdate(event) {
    this.setState({ item: event.target.value });
  }
  render() {
    return (
      <div className="NewTodo">
        <InputGroup>
          <Input
            type="text"
            onChange={this.handleUpdate}
            value={this.state.item}
            placeholder="Input item name here..."
          />
          <Button onClick={this.addTodo} color="primary">
            Add
          </Button>
        </InputGroup>
      </div>
    );
  }
}

export default NewTodo;
