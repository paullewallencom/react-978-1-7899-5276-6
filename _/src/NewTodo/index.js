import NewTodo from "./NewTodo";
export default NewTodo;
