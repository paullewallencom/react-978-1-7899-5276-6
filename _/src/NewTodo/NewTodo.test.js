import React from "react";
import ReactDOM from "react-dom";
import { shallow } from "enzyme";
import renderer from "react-test-renderer";

import NewTodo from "./NewTodo";

describe(NewTodo, () => {
  const mockAddTodo = jest.fn();
  const component = shallow(<NewTodo addTodo={mockAddTodo} />);

  it("renders without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(<NewTodo addTodo={mockAddTodo} />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it("renders and matches our snapshot", () => {
    const component = renderer.create(<NewTodo addTodo={mockAddTodo} />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("renders a Todo component", () => {
    expect(component.contains(<div className="NewTodo" />));
  });

  it("contains the form", () => {
    expect(component.find("Input")).toHaveLength(1);
    expect(component.find("Button")).toHaveLength(1);
  });

  it("calls the passed in addTodo function when add button is clicked", () => {
    component.find("Button").simulate("click");
    expect(mockAddTodo).toBeCalled();
  });

  it("updates the form when keys are pressed", () => {
    const updateKey = "New Todo";
    component.instance().handleUpdate({ target: { value: updateKey } });
    expect(component.state("item")).toEqual(updateKey);
  });

  it("blanks out the Todo Name when the button is clicked", () => {
    const updateKey = "I should be empty";
    component.instance().handleUpdate({ target: { value: updateKey } });
    expect(component.state("item")).toEqual(updateKey);
    component.find("Button").simulate("click");
    expect(component.state("item")).toHaveLength(0);
  });
});
