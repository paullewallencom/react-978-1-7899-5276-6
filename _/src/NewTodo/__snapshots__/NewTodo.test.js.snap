// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewTodo renders and matches our snapshot 1`] = `
<div
  className="NewTodo"
>
  <div
    className="input-group"
  >
    <input
      className="form-control"
      onChange={[Function]}
      placeholder="Input item name here..."
      type="text"
      value=""
    />
    <button
      aria-label={null}
      className="btn btn-primary"
      onClick={[Function]}
      type="button"
    >
      Add
    </button>
  </div>
</div>
`;
